server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestNodesHotThreadsAction.java

```diff
@@ -60,6 +82,11 @@ public String getName() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        List<String> deprecatedRoutesPath = deprecatedRoutes().stream().map(DeprecatedRoute::getPath).collect(Collectors.toList());
+        int getDeprecatedRoutesPathIndex = deprecatedRoutesPath.indexOf(request.rawPath());
```

Comment:
This shouldn't be needed as the REST infrastructure will handle for you when a deprecated path is invoked. Did you find some reason that you needed to add this?

Reply:
Because I want to set customized warning message for each deprecated route

Reply:
Besides, I want to call deprecationLogger to assert warnings when users use the deprecated path. Maybe I misinterpret something here...

Reply:
The REST infrastructure will do this for you, that's what you're passing to the `DeprecatedRoute` constructor.

Reply:
Oh I see. My main purpose of adding those lines is that I want to test them. If I delete those two lines, I cannot invoke `deprecationLogger.deprecated` to assert warnings.

Reply:
If there is no `deprecationLogger.deprecated` in `prepareRequest` method, I cannot test if the given request is the deprecated one, whether it will assert correct warning messages in `RestNodesHotThreadsActionTests.java`

Reply:
I am quite confused about how to get rid of these two lines while maining the tests in `RestNodesHotThreadsActionTests.java`

Reply:
I don’t think the tests are important because it’s basically testing the REST infrastructure which is tested elsewhere.

Reply:
I added these tests in `RestNodesHotThreadsActionTests.java` because in issue #52640, @DaveCTurner mentioned the need for adding tests for warnings. Maybe I need to add the kind of tests that @DaveCTurner mentioned in other places?

Reply:
Okay. You can add REST tests in `rest-api-spec`.

Reply:
Is there any reason that it is better for me to add tests by writing yml files instead of adding tests in RestNodesHotThreadsActionTests.java? @DaveCTurner mentioned the PR #50499 so I create tests in RestNodesHotThreadsActionTests.java. In that PR, the author create tests in `server/src/main/java/org/elasticsearch/rest/action/cat/RestNodesAction.java`. Thanks!

Reply:
I think the REST tests are more appropriate since this is about testing the deprecation of endpoints on the REST layer. One difference between the implementation here and in `RestNodesAction` as in the PR that you reference is that the deprecation is handled in different layers. Here, it's handled in the REST layer, there it's handled in the actual processing of the request (below the REST layer). That makes testing a little easier there. Here, you would have to wire up an entire REST layer (`RestController`) to get similar kind of coverage.

Yet, I still think that the REST tests are the best place to test REST layer changes.

Reply:
I found that adding tests in rest-api-spec is kind of complicated to catch the warning corresponding to different paths. So I removed all the tests! Thanks!